webserver/cors/src/main/java/io/helidon/webserver/cors/CrossOriginConfig.java

```diff
@@ -0,0 +1,308 @@
+        private boolean allowCredentials;
+        private long maxAge = DEFAULT_AGE;
+
+        private Builder() {
+        }
+
+        /**
+         * Creates a new builder based on the values in an existing {@code CrossOriginConfig} object.
+         *
+         * @param original the existing cross-origin config object
+         * @return new Builder initialized from the existing object's settings
+         */
+        public static Builder from(CrossOriginConfig original) {
+            return new Builder()
+                    .pathPrefix(original.pathPrefix)
+                    .enabled(original.enabled)
+                    .allowCredentials(original.allowCredentials)
+                    .allowHeaders(original.allowHeaders)
+                    .allowMethods(original.allowMethods)
+                    .allowOrigins(original.allowOrigins)
+                    .exposeHeaders(original.exposeHeaders)
+                    .maxAge(original.maxAge);
+        }
+
+        /**
+         * Creates a new {@code Builder}instance from the specified configuration.
+         *
+         * @param config node containing cross-origin information
+         * @return new {@code Builder} initialized from the config
+         */
+        public static Builder from(Config config) {
+            return Loader.Basic.builder(config);
+        }
+
+        @Override
+        public Builder apply(Config config) {
+            return from(config);
+        }
+
+        /**
+         * Updates the path prefix for this cross-origin config.
+         *
+         * @param pathPrefix new path prefix
+         * @return updated builder
+         */
+        public Builder pathPrefix(String pathPrefix) {
+            this.pathPrefix = pathPrefix;
+            return this;
+        }
+
+        String pathPrefix() {
+            return pathPrefix;
+        }
+
+        @Override
+        public Builder enabled(boolean enabled) {
+            this.enabled = enabled;
+            return this;
+        }
+
+        @Override
+        public Builder allowOrigins(String... origins) {
+            this.origins = copyOf(origins);
+            return this;
+        }
+
+        @Override
+        public Builder allowHeaders(String... allowHeaders) {
+            this.allowHeaders = copyOf(allowHeaders);
+            return this;
+        }
+
+        @Override
+        public Builder exposeHeaders(String... exposeHeaders) {
+            this.exposeHeaders = copyOf(exposeHeaders);
+            return this;
+        }
+
+        @Override
+        public Builder allowMethods(String... allowMethods) {
+            this.allowMethods = copyOf(allowMethods);
+            return this;
+        }
+
+        @Override
+        public Builder allowCredentials(boolean allowCredentials) {
+            this.allowCredentials = allowCredentials;
+            return this;
+        }
+
+        @Override
+        public Builder maxAge(long maxAge) {
```

Comment:
Whenever time units are used and are not as a parameter, please name the method accordingly (e.g. `maxAgeSeconds` or `maxAgeMillis`). You can also add another method that accepts a time unit.